### ISSUE TITLE
Using pollInterval causing issues with updating components

### ISSUE THREAD

**Issue by clayne11** [apollographql/react-apollo#789]
# Issue

When I use `pollInterval` on a query, mutations that change that entity after the fact no longer update the React component.

## Steps to Reproduce

1. Set up a query using `pollInterval`.
1. Wait for the interval to expire at least once so that the query has been re-fetched.
1. Perform a mutation to update data that is being fetch and rendered by a React component.
1. The store will update properly, but the component will not re-render.

**Comment by helfer** [comment]
@clayne11 thanks for the bug report, could you provide a reproduction of this using the [error-template](https://github.com/apollographql/react-apollo-error-template)? That would help a lot.

**Comment by clayne11** [comment]
Turns out the issue was unrelated but extremely difficult to track down.

The issue ended up actually being related to https://github.com/apollographql/apollo-client/issues/819.

I have 3 different polling queries occurring on one page. When all of the polling queries were turned on the error was not being printed in the console at all and must have been getting eaten up somewhere inside of apollo-client.

I had to turn on one of the three polling queries, turn off the other two, and perform some seemingly unrelated actions before getting this error to fire.

I've also run into this same issue multiple other times while trying to roll out a new feature.
